x/bank/src/keeper.rs

```diff
@@ -567,12 +565,15 @@ impl<SK: StoreKey, PSK: ParamsSubspaceKey, AK: AuthKeeper<SK, M>, M: Module>
 
             from_balance.amount -= send_coin.amount;
 
-            from_account_store.set(
-                send_coin.denom.clone().to_string().into_bytes(),
-                from_balance.encode_vec(),
-            )?;
-
-            //TODO: if balance == 0 then denom should be removed from store
+            // if balance == 0 then denom should be removed from store
+            if from_balance.amount.is_zero() {
+                from_account_store.delete(send_coin.denom.to_string().as_bytes())?;
```

Comment:
I think that I saw this. But where? /s

Reply:
[This](https://github.com/rumos-io/gears/pull/190) PR was for a different part of the code.